CONTRIBUTING.md

```diff
@@ -0,0 +1,80 @@
+# GlusterFS project Contribution guidelines
+
+## Development Workflow
+
+0. Fork Repository
+   - Visit [GlusterFS repository](https://github.com/gluster/glusterfs) and click 'Fork'.
+
+1. Clone Repository
+   - Clone the glusterfs repo freshly from github using below steps.
+
+```
+   git clone git@github.com:${username}/glusterfs.git
+   cd glusterfs/
+   git remote add upstream git@github.com:gluster/glusterfs.git
+```
+
+   - We follow most of the details as per the [document here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests).
+
+2. Commit Message / PR description:
+   - The name of the branch on your personal fork can start with issueNNNN, followed by anything of your choice.
+   - PRs continue to have the title of format "component: <title>", like it is practiced now.
+   - When you open a PR, having a reference Issue for the commit is mandatory in GlusterFS.
+   - Commit message can have, either `Fixes: #NNNN` or `Updates: #NNNN` in a separate line in the commit message.
+     - Here, NNNN is the Issue ID in glusterfs repository.
+   - Each commit needs the author to have the "Signed-off-by: Name<email>" line.
+     - Can do this by `-s` option for `git commit`.
+   - If the PR is not ready for review, apply the label `work-in-progress`.
+     - Check the availability of "Draft PR" is present for you, if yes, use that instead.
+
+3. Tests:
+   - All the required smoke tests would be auto-triggered.
+   - The "regression" tests would be triggered by a comment **"/ok-to-test"** from anyone in the ['@gluster-all'](https://github.com/orgs/gluster/teams/gluster-all) group.
+
+4. Review Process:
+   - `+2` : is equivalent to "Approve" from the people in the maintainer's group.
+   - `+1` : can be given by a maintainer/reviewer by explicitly stating that in the comment.
+   - `-1` : provide details on required changes and pick "Request Changes" while submitting your review.
+   - `-2` : done by adding the "DO-NOT-MERGE" label.
+
+   - Any further discussions can happen as comments in the PR.
+
+5. Making changes:
+  - There are 2 approaches to submit changes done after addressing review comments.
+    - Commit changes as a new commit on top of the original commits in the branch, and push the changes to same branch (issueNNNN)
+    - Commit changes into the same commit with `--amend` option, and do a push to the same branch with `--force` option.
+
+6. Merging:
```

Comment:
Merge type is "Squash and Merge"?

Reply:
Yes. Only that is permitted. Will update the document

Reply:
Done